orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/servergroup/WaitForCapacityMatchTask.groovy

```diff
@@ -43,6 +43,11 @@ class WaitForCapacityMatchTask extends AbstractInstancesCheckTask {
     if (!serverGroup.capacity || serverGroup.capacity.desired != instances.size()) {
       return false
     }
+
+    if (stage.context.get("capacity")?.get("desired") != instances.size()) {
```

Comment:
If this key is missing, the stage will fail every time

Reply:
you mean context?